extensions/blocks/donations/donations.php

```diff
@@ -40,29 +94,170 @@ function register_block() {
+		if ( count( $donations ) > 1 ) {
+			if ( ! $nav ) {
+				$nav .= '<div class="donations__nav">';
+			}
+			$nav .= sprintf(
+				'<div role="button" tabindex="0" class="donations__nav-item" data-interval="%1$s">%2$s</div>',
+				esc_attr( $interval ),
+				$donation['title']
+			);
+		}
+		$headings .= sprintf(
+			'<h4 class="%1$s">%2$s</h4>',
+			esc_attr( $donation['class'] ),
+			$donation['heading']
+		);
+		$amounts  .= sprintf(
+			'<div class="donations__amounts %s">',
+			esc_attr( $donation['class'] )
+		);
+		foreach ( $donation['amounts'] as $amount ) {
+			$amounts .= sprintf(
+				'<div class="donations__amount" data-amount="%1$s">%2$s</div>',
+				esc_attr( $amount ),
+				esc_html( \Jetpack_Currencies::format_price( $amount, $currency ) )
+			);
+		}
+		$amounts    .= '</div>';
+		$extra_text .= sprintf(
+			'<p class="%1$s">%2$s</p>',
+			esc_attr( $donation['class'] ),
+			$donation['extraText']
+		);
+		$buttons    .= sprintf(
+			'<a class="wp-block-button__link donations__donate-button %1$s" href="%2$s">%3$s</a>',
+			esc_attr( $donation['class'] ),
+			esc_url( \Jetpack_Memberships::get_instance()->get_subscription_url( $plan_id ) ),
+			$donation['buttonText']
+		);
+	}
+	if ( $nav ) {
+		$nav .= '</div>';
+	}
+
+	$custom_amount = '';
+	if ( $attr['showCustomAmount'] ) {
+		$custom_amount        .= sprintf(
+			'<p>%s</p>',
+			$attr['customAmountText']
+		);
+		$default_custom_amount = \Jetpack_Memberships::SUPPORTED_CURRENCIES[ $currency ] * 100;
+		$custom_amount        .= sprintf(
+			'<div class="donations__amount donations__custom-amount">
+				%1$s
+				<div class="donations__amount-value" data-currency="%2$s" data-empty-text="%3$s"></div>
+			</div>',
+			esc_html( \Jetpack_Currencies::CURRENCIES[ $attr['currency'] ]['symbol'] ),
+			esc_attr( $attr['currency'] ),
+			esc_attr( \Jetpack_Currencies::format_price( $default_custom_amount, $currency, false ) )
+		);
 	}
 
-	return $content;
+	return sprintf(
+		'
+<div class="%1$s">
+	<div class="donations__container">
+	%2$s
+	<div class="donations__content">
+		<div class="donations__tab">
+			%3$s
+			<p>%4$s</p>
+			%5$s
+			%6$s
+			<hr class="donations__separator">
+			%7$s
+			%8$s
+		</div>
+	</div>
+</div>
+',
+		esc_attr( $classes ),
+		$nav,
+		$headings,
+		$attr['chooseAmountText'],
+		$amounts,
+		$custom_amount,
+		$extra_text,
+		$buttons
+	);
+}
+
+/**
+ * Determine if AMP should be disabled on posts having Donations blocks.
+ *
+ * @param bool    $skip Skipped.
+ * @param int     $post_id Post ID.
+ * @param WP_Post $post Post.
+ *
+ * @return bool Whether to skip the post from AMP.
+ */
+function amp_skip_post( $skip, $post_id, $post ) {
+	// When AMP is on standard mode, there are no non-AMP posts to link to where the donation can be completed, so let's
+	// prevent the post from being available in AMP.
+	if ( function_exists( 'amp_is_canonical' ) && \amp_is_canonical() && has_block( 'jetpack/donations', $post->post_content ) ) {
```

Comment:
Maybe we can use `BLOCK_NAME` here?

```suggestion
	if ( function_exists( 'amp_is_canonical' ) && \amp_is_canonical() && has_block( BLOCK_NAME, $post->post_content ) ) {
```

(I didn't think of that for the Pay with Paypal block, but I suppose we could do the same there as well).